src/test/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryIT.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.plugin.query;
+
+import static org.opensearch.neuralsearch.plugin.TestUtils.createRandomVector;
+import static org.opensearch.neuralsearch.plugin.TestUtils.objectToFloat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+import org.junit.Before;
+import org.opensearch.common.Strings;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.MatchAllQueryBuilder;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.neuralsearch.common.BaseNeuralSearchIT;
+
+import com.google.common.primitives.Floats;
+
+public class NeuralQueryIT extends BaseNeuralSearchIT {
+    private static final String TEST_BASIC_INDEX_NAME = "test-neural-basic-index";
+    private static final String TEST_MULTI_FIELD_INDEX_NAME = "test-neural-multi-field-index";
+    private static final String TEST_NESTED_INDEX_NAME = "test-neural-nested-index";
+    private static final String TEST_QUERY_TEXT = "Hello world";
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_1 = "test-knn-vector-1";
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_2 = "test-knn-vector-2";
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_NESTED = "nested.knn.field";
+
+    private static final int TEST_DIMENSION = 384;
+    private static final SpaceType TEST_SPACE_TYPE = SpaceType.L2;
+    private final float[] testVector = createRandomVector(TEST_DIMENSION);
+    private final AtomicReference<String> modelId = new AtomicReference<>();
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        modelId.compareAndSet(null, prepareModel());
+    }
+
+    private void maybeInitializeIndex(String indexName) throws IOException {
```

Comment:
[nit-pick] move this private function after all the public and protected functions.